app.py

```diff
@@ -146,7 +146,12 @@ def package_releases_update(
     if only_if_package_seen:
         # An optimization - we don't need to iterate over a large set present on index.
         # Check only packages known to Thoth.
-        package_names = graph.get_python_packages()
+        package_names = list(
+            set(
+                [p_name[0] for p_name in graph.get_python_packages_all(
+                    count=graph.get_python_packages_count_all())]
```

Comment:
Can we do this on the database side? Introduce something like `graph.get_python_package_names(distinct=True)`.

Reply:
Yes we can

Reply:
Implemented
```
    def get_python_package_names_all(
        self,
        *,
        os_name: str = None,
        os_version: str = None,
        python_version: str = None,
        distinct: bool = False,
    ) -> List[str]:
        """Retrieve unique Python Packages known by Thoth.

        Examples:
        >>> from thoth.storages import GraphDatabase
        >>> graph = GraphDatabase()
        >>> graph.get_python_packages_names_all()
        ['regex', 'tensorflow']
        """
```

```get_python_package_names_all()
['setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'pip', 'pip', 'pip', 'pip', 'pip', 'pip', 'pip', 'pip', 'pip', 'pip', 'pip', 'pip', 'pip', 'pip', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools', 'setuptools']
get_python_package_names_all(distinct=True)
['setuptools', 'pip']
```